lib/SyTest/Federation/Server.pm

```diff
@@ -225,6 +226,67 @@ sub on_request_key_v2_server
    } ) );
 }
 
+sub on_request_federation_v1_query_directory
+{
+   my $self = shift;
+   my ( $req, $alias ) = @_;
+
+   my $room_id = $self->{datastore}->get_alias( $alias ) or
+      return Future->done( response => HTTP::Response->new(
+         404, "Not found", [ Content_length => 0 ], "",
```

Comment:
The spec is rather vague on the subject - neither the word `failure` nor `error` appears at all in http://matrix.org/docs/spec/r0.0.0/server_server.html
Empirically it is observed to be happy with empty-length errors.